specifyweb/frontend/js_src/lib/components/QueryBuilder/ToForms.tsx

```diff
@@ -34,10 +35,16 @@ export function QueryToForms({
       ? (results[index]![queryIdField] as number)
       : Array.from(selectedRows)[index];
 
+  const filteredResults = f.maybe(results, filterArray);
```

Comment:
1. you don't need the actual results, you just want to know if any results are not undefined, right? in that case, think about how you could use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some in here to improve performance and maybe even simplify code a bit
2. you could use useMemo